src/js/typescript.js

```diff
@@ -62,7 +62,7 @@ export default class TypeScriptCompiler extends SimpleCompilerBase {
       fileName: filePath.match(/\.(ts|tsx)$/i) ? path.basename(filePath) : null
     };
 
-    if (isTsx && options.builtinOpts.hotModuleReload !== false) {
+    if (isTsx && options.builtinOpts.hotModuleReload === true) {
```

Comment:
We should enable this by default, there's (this bug nonwithstanding) no reason not to